[OptimistLabyrinth] 🚀 1단계 - 서비스 구성하기

OptimistLabyrinth opened the pull request:
미션 완료 후

- Bastion 인스턴스(EC2 name: optimistlabyrinth-bastion-server-1st)
- Service 인스턴스(EC2 name: optimistlabyrinth-service-instance-1st) 

둘 다 정지했습니다.

Commit:
feat: 🚀 1단계 - 서비스 구성하기

1단계 과제 완료 후
웹사이트 방문 화면 캡쳐 : ${projectRoot}/photo-capture/phase-01.png

OptimistLabyrinth commented:
<img width="1080" alt="phase-01" src="https://user-images.githubusercontent.com/16411622/203963874-6ae64841-dfa8-440d-974f-b13abc9cc97b.png">

joojimin reviewed (changes requested):
안녕하세요~ 기성님
미션 요구사항을 다시한번 점검해보셔야할거같아요~!
CIDR 규칙, 서비스넷생성 규칙 전반적으로 다시한번 점검부탁드립니다

Commit:
feat: EC2 인스턴스 재시작 후 다시 PR 요청하기

EC2 인스턴스는 정지하고 재시작할때마다 Public IP 가 바뀌어서
기존에 설정해둔 보안 규칙이 제대로 동작하지 않게 된다.

OptimistLabyrinth commented:
바스티온 서버 인스턴스(optimistlabyrinth-bastion-server-1st) 로 ssh 접속하신 후 터미널에서 
```bash
ssh ubuntu@service-instance
``` 
라고 입력하시면
서비스 배포 인스턴스(optimistlabyrinth-service-instance-1st) 에 접속하게 됩니다.
`/home/ubuntu/nextstep` 디렉토리에 배포중인 서비스 프로세스 아이디를 pid_NNNN 이라고 기록해두었습니다.

joojimin reviewed (changes requested):
안녕하세요 기성님
미션 요구사항을 다시 살펴보셔야할 것 같아요!
기존의 생성되어있던 VPC를 사용하신걸까요?? 사용하신 VPC, Subnet CIDR 규칙이 요구사항에 부합하지않네요 ㅠㅠ
추가로 내부망 인스턴스 미생성 및 보안그룹 설정이 실제IP로 픽스되어 설정되어있는 것 같습니다!
천천히 다시한번 점검 부탁드려요~!

Commit:
feat: 미션 진행 페이지에 있는대로 1단계 실행하기

세 개의 망이 필요하다: 외부망, 내부망, 관리망

VPC 생성 -> Subnet 생성 -> Internet Gateway 연결 -> Route Table 생성
-> Security Group 생성 -> EC2 인스턴스 생성

internet gateway 가 외부와 동작하도록 하려면
route table 에 0.0.0.0/0 패턴을 추가해야 한다

OptimistLabyrinth commented:
미션 요구사항을 차근차근 다시 분석해서 하나씩 실행했습니다.
다시 리뷰 요청드릴께요.

joojimin reviewed (changes requested):
안녕하세요~ 기성님!
꼼꼼하게 다시 설정 잘해주셨는데요! 
생성하신 모든 서브넷이 하나의 라우팅테이블에 연결되어있네요!
해당 라우팅테이블은 인터넷게이트웨이가 설정되어있는데 모든 서브넷에 IGW가 필요한지 체크부탁드려요~

OptimistLabyrinth commented:
올바른 지적이십니다.
수정 후 다시 리뷰 요청드립니다.

![photo-route-tables](https://user-images.githubusercontent.com/16411622/207002525-1f8bd1ba-9447-4a04-a6b8-e675b756c826.png)

joojimin reviewed (approved):
미션진행 고생하셨습니다 기성님~!!
설정 잘 마무리해주셨네요! 다음미션도 화이팅입니다!!
